cekit/descriptor/packages.py

```diff
@@ -69,28 +71,34 @@ def __init__(self, descriptor):
             descriptor['filename'] = '%s.repo' % descriptor['name'].replace(' ', '_')
 
         if not (('url' in descriptor) ^
-                ('odcs' in descriptor) ^
                 ('id' in descriptor) ^
-                ('rpm' in descriptor)):
+                ('rpm' in descriptor) ^
+                ('content_sets' in descriptor) ^
+                ('content_sets_file' in descriptor)):
             raise CekitError("Repository '%s' is invalid, you can use only one of "
-                             "['id', 'odcs', 'rpm', 'url']"
+                             "['id', 'content_sets_file', 'content_sets', 'rpm', 'url']"
                              % descriptor['name'])
 
-        if tools.cfg['common']['redhat'] and 'id' in descriptor:
-            descriptor['odcs'] = {'pulp': descriptor['id']}
-            del descriptor['id']
+        if ('content_sets_file' in descriptor and 'content_sets' in descriptor):
+            raise CekitError("You cannot specify content_sets and content_sets_file together!")
 
         if 'url' not in descriptor:
             descriptor['url'] = {}
 
+        if 'content_sets_file' in descriptor:
+            with open(descriptor['content_sets_file'], 'r') as file_:
```

Comment:
We read it twice? It is read in `osbs.py` too.

Reply:
no its not.. in osbs.py container.yaml is read not contentsets.yml